libraries/ts-command-line/src/providers/ScopedCommandLineAction.ts

```diff
@@ -0,0 +1,231 @@
+   * must be defined with this group name.
+   */
+  public static ScopingParameterGroupName: 'scoping' = 'scoping';
+
+  public constructor(options: ICommandLineActionOptions) {
+    super(options);
+
+    this._options = options;
+    this._scopingParameters = [];
+  }
+
+  /**
+   * {@inheritDoc CommandLineParameterProvider.parameters}
+   */
+  public get parameters(): ReadonlyArray<CommandLineParameter> {
+    if (this._scopedCommandLineParser) {
+      return [...super.parameters, ...this._scopedCommandLineParser.parameters];
+    } else {
+      return super.parameters;
+    }
+  }
+
+  /**
+   * {@inheritdoc CommandLineAction._processParsedData}
+   * @internal
+   */
+  public _processParsedData(parserOptions: ICommandLineParserOptions, data: ICommandLineParserData): void {
+    // override
+    super._processParsedData(parserOptions, data);
+
+    this._unscopedParserOptions = parserOptions;
+
+    // Generate the scoped parser using the parent parser information. We can only create this after we
+    // have parsed the data, since the parameter values are used during construction.
+    this._scopedCommandLineParser = new InternalScopedCommandLineParser({
+      ...parserOptions,
+      actionOptions: this._options,
+      unscopedActionParameters: this.parameters,
+      onDefineScopedParameters: this.onDefineScopedParameters.bind(this),
+      onExecute: this.onExecute.bind(this)
+    });
+  }
+
+  /**
+   * {@inheritdoc CommandLineAction._execute}
+   * @internal
+   */
+  public async _execute(): Promise<void> {
+    // override
+    if (!this._unscopedParserOptions || !this._scopedCommandLineParser) {
+      throw new Error('The CommandLineAction parameters must be processed before execution.');
+    }
+    if (!this.remainder) {
+      throw new Error('CommandLineAction.onDefineParameters must be called before execution.');
+    }
+
+    // The '--' argument is required to separate the action parameters from the scoped parameters,
+    // so it needs to be trimmed. If remainder values are provided but no '--' is found, then throw.
+    const scopedArgs: string[] = [];
+    if (this.remainder.values.length) {
+      if (this.remainder.values[0] !== '--') {
+        // Immitate argparse behavior and log out usage text before throwing.
+        console.log(this.renderUsageText());
+        throw new CommandLineParserExitError(
+          // argparse sets exit code 2 for invalid arguments
+          2,
+          // model the message off of the built-in "unrecognized arguments" message
+          `${this._unscopedParserOptions.toolFilename} ${this.actionName}: error: Unrecognized ` +
+            `arguments: ${this.remainder.values[0]}.`
+        );
+      }
+      scopedArgs.push(...this.remainder.values.slice(1));
+    }
+
+    // Call the scoped parser using only the scoped args.
+    await this._scopedCommandLineParser.executeWithoutErrorHandling(scopedArgs);
```

Comment:
If this isn't `-h`/`--help`, seems like we should be able to just run `await this.onExecute()` right after this line.